SparkPost/matchbox
# FE-825 - Update Modal Component

[FE-825](https://jira.int.messagesystems.com/browse/FE-825)

### What Changed
- Refactors Modal component to use functional instead of class components
- Removes CSS modules in favor of Styled Components
- Removes distinct `<Content/>` component in favor of a private, `<ModalContent/>` component
- Adds supporting unit tests
- Adds mock for `react-transition-group` for use by Jest
- Adds supporting end-to-end tests
- Updates Storybook examples to match mocks, refactors Storybook `<ModalDemo/>` component from a class to a function

### How To Test or Verify
- Run `npm run start:storybook` and verify the modal renders according to the mocks
- Verify the animation works as expected
- Verify unit/e2e tests adequately test the component
- Verify a11y attributes survived the update in their appropriate locations
- Verify the modal is controllable via a keyboard

### PR Checklist
